ouster_ros/src/ros.cpp

```diff
@@ -119,20 +120,49 @@ void scan_to_cloud(const ouster::XYZLut& xyz_lut,
         read_and_cast(), reflectivity);
 
     auto points = ouster::cartesian(range, xyz_lut);
-
-    for (auto u = 0; u < ls.h; u++) {
-        for (auto v = 0; v < ls.w; v++) {
-            const auto xyz = points.row(u * ls.w + v);
-            const auto ts = (ls.header(v).timestamp - scan_ts).count();
-            cloud(v, u) = ouster_ros::Point{
-                {{static_cast<float>(xyz(0)), static_cast<float>(xyz(1)),
-                  static_cast<float>(xyz(2)), 1.0f}},
-                static_cast<float>(signal(u, v)),
-                static_cast<uint32_t>(ts),
-                static_cast<uint16_t>(reflectivity(u, v)),
-                static_cast<uint8_t>(u),
-                static_cast<uint16_t>(near_ir(u, v)),
-                static_cast<uint32_t>(range(u, v))};
+    if (filter.enabled) {
+        std::cout << "1) ";
+        float x, y, z;
+        for (auto u = 0; u < ls.h; u++) {
+            for (auto v = 0; v < ls.w; v++) {
+                const auto xyz = points.row(u * ls.w + v);
+                const auto ts = (ls.header(v).timestamp - scan_ts).count();
+                // filter
+                x = xyz(0);
+                y = xyz(1);
+                z = xyz(2);
+                if (((x > filter.val_min[0]) && (x < filter.val_max[0])) &&
+                    ((y > filter.val_min[1]) && (y < filter.val_max[1])) &&
+                    ((z > filter.val_min[2]) && (z < filter.val_max[2]))) {
+                    x = 99999.0;
```

Comment:
I see it is a exclusion filter, perhaps the parameter names can be modified so that that is directly apparent? Also, does setting the xyz values make it a NaN point?